File: Api/Models/Book.cs
namespace Api.Models;

public class Book : BaseModel
{
    public int Id { get; set; }
    public required string Title { get; set; } = default!;
    public required string Author { get; set; } = default!;
    public required DateOnly PublishedDate { get; set; }
    public List<Quote> Quotes { get; set; } = new();
}

File: client/src/types/book.ts
export type Book = {
  id: number;
  title: string;
  author: string;
  publishedDate: string;
  quotes?: any[];
  createdBy: string;
  createdAt: string;
  updatedBy: string;
  updatedAt: string;
};

export type BookCreateUpdateModel = {
  id: number;
  title: string;
  author: string;
  publishedDate: string;
};
